src/izgbs.py

```diff
@@ -67,17 +82,18 @@ def create_hypotheses_df(num_h):
         Returns:
             TODO
     '''
-    # This dataframe will contain 1 record per machine count
-    res_cols = ['Machines', 'Feasible', 'Avg', 'MaxAvg']
-    # Create an empty dataframe the same size as the locations dataframe
-    voter_cols = np.zeros((num_h, len(res_cols)))
-    hyp_results = pd.DataFrame(voter_cols, columns=res_cols)
-    # Populates the Machine count field
-    hyp_results['Machines'] = (hyp_results.index + 1).astype('int')
+    hyp_results = {}
+    i = 0
+    # the index should have the same value of Machines
+    while i < num_h:
+        index = i + 1
+        res_dict = initialize_result_dict(i + 1)
+        hyp_results[index] = res_dict
```

Comment:
You don't need the initialize_result_dict() function. You can instead directly create the dict here.
Ex:
```python
res_dict = {
   'Machines': i + 1,
   'Feasible': False,
   'BatchAvg': 0,
   'BatchMaxAvg': 0
}
```